plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlIntegrationSmokeTest.java

```diff
@@ -427,6 +429,168 @@ public void testAggregationPushdown()
+                ImmutableList.of("2, 2, 2, 2", "4, 4, 4, 4"))) {
+            assertThat(query("SELECT covar_pop(t_double1, t_double2), covar_pop(t_real1, t_real2) FROM " + testTable.getName())).isFullyPushedDown();
+            assertThat(query("SELECT covar_samp(t_double1, t_double2), covar_samp(t_real1, t_real2) FROM " + testTable.getName())).isFullyPushedDown();
+        }
+
+        // non-whole number results
+        try (TestTable testTable = new TestTable(
+                postgreSqlServer::execute,
+                getSession().getSchema().orElseThrow() + ".test_covariance_pushdown",
+                "(t_double1 DOUBLE PRECISION, t_double2 DOUBLE PRECISION, t_real1 REAL, t_real2 REAL)",
+                ImmutableList.of("1, 2, 1, 2", "100000000.123456, 4, 100000000.123456, 4", "123456789.987654, 8, 123456789.987654, 8"))) {
+            assertThat(query("SELECT covar_pop(t_double1, t_double2), covar_pop(t_real1, t_real2) FROM " + testTable.getName())).isFullyPushedDown();
+            assertThat(query("SELECT covar_samp(t_double1, t_double2), covar_samp(t_real1, t_real2) FROM " + testTable.getName())).isFullyPushedDown();
+        }
+    }
+
+    @Test
+    public void testCorrPushdown()
+    {
+        // empty table
+        try (TestTable testTable = new TestTable(
+                postgreSqlServer::execute,
+                getSession().getSchema().orElseThrow() + ".test_corr_pushdown",
+                "(t_double1 DOUBLE PRECISION, t_double2 DOUBLE PRECISION, t_real1 REAL, t_real2 REAL)")) {
+            assertThat(query("SELECT corr(t_double1, t_double2), corr(t_real1, t_real2) FROM " + testTable.getName())).isFullyPushedDown();
+        }
+
+        // test some values for which the aggregate functions return whole numbers
+        try (TestTable testTable = new TestTable(
+                postgreSqlServer::execute,
+                getSession().getSchema().orElseThrow() + ".test_corr_pushdown",
+                "(t_double1 DOUBLE PRECISION, t_double2 DOUBLE PRECISION, t_real1 REAL, t_real2 REAL)",
+                ImmutableList.of("2, 2, 2, 2", "4, 4, 4, 4"))) {
+            assertThat(query("SELECT corr(t_double1, t_double2), corr(t_real1, t_real2) FROM " + testTable.getName())).isFullyPushedDown();
+        }
+
+        // non-whole number results
+        try (TestTable testTable = new TestTable(
+                postgreSqlServer::execute,
+                getSession().getSchema().orElseThrow() + ".test_corr_pushdown",
+                "(t_double1 DOUBLE PRECISION, t_double2 DOUBLE PRECISION, t_real1 REAL, t_real2 REAL)",
+                ImmutableList.of("1, 2, 1, 2", "100000000.123456, 4, 100000000.123456, 4", "123456789.987654, 8, 123456789.987654, 8"))) {
+            assertThat(query("SELECT corr(t_double1, t_double2), corr(t_real1, t_real2) FROM " + testTable.getName())).isFullyPushedDown();
+        }
+    }
+
+    @Test
+    public void testRegrPushdown()
```

Comment:
let's  testRegrPushdown -> testRegrAggregationPushdown (same for others , also those existing ones)

-- as we're getting more test methods like that, this naming pattern will make them easier to find